### PR TITLE
ci: use Container-Optimized OS public image on the VM

### LINKED ISSUE #5618: ci: compute engine VMs have not been using the expected OS (debian-11)
## Motivation

We've been pulling the Debian image the wrong way, as this was being used as a container but it was meant to be the VM image. And the image being pulled to create the internal container has been causing crashes as this images do not exists on Google's container repositories.

### Specifications

https://cloud.google.com/sdk/gcloud/reference/compute/instances/create-with-container#--image-family and https://cloud.google.com/compute/docs/images/os-details#debian

### Designs

Add `--image-project=debian-cloud` and `--image-family=debian-11` as stated in the official documentation: https://cloud.google.com/sdk/gcloud/reference/compute/instances/create-with-container#--image-project

## Related Work

#5602 and #5615

### PR DESCRIPTION
**Pull request by gustavovalverde**
## Previous behavior:

We were pulling the debian image the wrong way, as this was being used as a container but it was meant to be the VM image

The image being pulled to create the internal container has been causing crashes as this images do not exists on Google's container repositories

Fixes #5618

## Expected behavior:

Use a public image as ~debian-11~ `cos-stable`  to get multiple benefits from it, as being able to use machine-images (#5615) ~and automatic disk resizing (which is now possible as we're using COS images, but those are more restrictive)~. We would have preferred `debian-11` but the host image does not include Docker. 

## Solution

Add `--image-project=cos-cloud` and `--image-family=cos-stable` as stated in the official documentation: https://cloud.google.com/sdk/gcloud/reference/compute/instances/create-with-container#--image-project

More info: https://cloud.google.com/compute/docs/images/os-details#import

## Review

Anyone from @ZcashFoundation/devops-reviewers 

### Reviewer Checklist

  - [x] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [x] Are the PR labels correct?
  - [x] Does the code do what the ticket and PR says?
  - [ ] How do you know it works? Does it have tests?

## Follow Up Work

This might reduce some error's we've been having with GCP

### CODE REVIEW

**Comment by gustavovalverde** [comment]
Docker is not available inside the official public image 🙃

**Comment by dconnolly** [comment]
@mergify update

**Comment by dconnolly** [comment]
@gustavovalverde this is in draft, is this still critical?

**Comment by gustavovalverde** [comment]
@mergifyio update

**Review by arya2 — APPROVED** [review]
Looks good!

PR title may need an update.

File: .github/workflows/continous-delivery.yml [hunk 1]
```diff
@@ -120,6 +120,8 @@ jobs:
         run: |
           gcloud compute instance-templates create-with-container zebrad-${{ needs.versioning.outputs.major_version || env.GITHUB_REF_SLUG_URL }}-${{ env.GITHUB_SHA_SHORT }} \
           --boot-disk-type=pd-ssd \
+          --image-project=cos-cloud \
+          --image-family=cos-stable \
```
Comment (dconnolly) [inline .github/workflows/continous-delivery.yml:124]: Once upon a time I could have sworn this was the default choice, maybe that changed

Comment (arya2) [reply .github/workflows/continous-delivery.yml:124]: I think it's changed:

> By default, debian-11 is assumed for this flag.

https://cloud.google.com/sdk/gcloud/reference/compute/instances/create-with-container#--image-family

**Review by dconnolly — APPROVED** [review]

**Pull request merged**